app/aws/Federation.scala

```diff
@@ -38,32 +38,42 @@ object Federation {
     * default, which may depend on their timezone.
     *
     * The tests explain the different cases, this is a tricky function.
+    *
+    * @param permission
+    *   Used to determine if the session duration should be short-term or
+    *   standard.
+    * @param requestedDuration
+    *   Optionally, a user can request a duration for their session, which may
+    *   or may not be honoured.
+    * @param optClock
+    *   [[Clock]] to use to determine the current time.
     */
   def duration(
       permission: Permission,
-      requestedSeconds: Option[Duration] = None,
-      timezone: Option[DateTimeZone] = None
+      requestedDuration: Option[Duration] = None,
+      optClock: Option[Clock] = None
```

Comment:
Now passing in an optional [Clock](https://dev.java/learn/date-time/clock/) instead so that the intention of the method still makes sense but we are also able to test it easily in different scenarios.  
A Clock represents a timezone but within that timezone the time can either be passing at a real-world rate or fixed.  A fixed clock always gives the same value for current time.  
According to the javadocs:
> Best practice for applications is to pass a Clock into any method that requires the current instant and time-zone.

We aren't doing this generally but it can be revisited elsewhere in a later PR.